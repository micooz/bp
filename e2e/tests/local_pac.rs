use bp_core::{Address, ClientOptions, Options};
use cmd_lib::run_fun;
use e2e::runner::run_bp_custom;

#[tokio::test(flavor = "multi_thread")]
async fn test_pac() {
    let pac_bind: Address = "127.0.0.1:8000".parse().unwrap();

    let opts = Options::Client(ClientOptions {
        pac_bind: Some(pac_bind.clone()),
        acl: Some("tests/fixtures/acl.txt".to_string()),
        ..Default::default()
    });

    run_bp_custom(opts, Some("127.0.0.1"), Some(3000)).await;

    let pac_bind = pac_bind.to_string();

    insta::assert_snapshot!("proxy.pac", run_fun!(curl $pac_bind/proxy.pac).unwrap());
    insta::assert_snapshot!("proxy.pac2", run_fun!(curl $pac_bind/proxy.pac?xxx).unwrap());
}
