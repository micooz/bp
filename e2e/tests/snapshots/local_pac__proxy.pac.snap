---
source: e2e/tests/local_pac.rs
assertion_line: 19
expression: run_fun!(curl $pac_bind / proxy.pac).unwrap()

---
// This file is automatically generated.

function FindProxyForURL(url, host) {
  // extract port number from url
  var port_match = url.match(/:(\d{0,5})/);
  var port = port_match[1] ? Number(port_match[1]) : null;

  if (shExpMatch(host, "*baidu.com*")) return "PROXY 127.0.0.1:3000";
  // #baidu.com
  if (host === "cn.bing.com") return "PROXY 127.0.0.1:3000";

  return "DIRECT";
}
